### PR TITLE
aws_api_gateway_deployment

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_api_gateway_deployment.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  desc 'Gets information about a Deployment resource.'
+
+  example "
+    describe aws_api_gateway_deployment(rest_api_id: 'rest_api_id', deployment_id: 'deployment_id') do
+      it { should exist }
+    end
+  "
+
+  def initialize(opts = {})
+    opts = { rest_api_id: opts, deployment_id: opts } if opts.is_a?(String)
```
Comment (SathishAchilles) [inline libraries/aws_api_gateway_deployment.rb:16]: what are we doing here?

File: docs/resources/aws_api_gateway_deployment.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+---
```
Comment (SathishAchilles) [inline docs/resources/aws_api_gateway_deployment.md:1]: please add it to readme

File: docs/resources/aws_api_gateway_deployment.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+Use `should` to check if the work_group name is available.
+
+    describe aws_api_gateway_deployment(rest_api_id: 'value', deployment_id: 'value') do
+      it { should be_available }
+    end
+
+## AWS Permissions
+
+Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `apigateway_client.get_deployment` action with `Effect` set to `Allow`.
```
Comment (SathishAchilles) [inline docs/resources/aws_api_gateway_deployment.md:81]: is there a role like this `apigateway_client.get_deployment` ? Could you please a point a document to this role?

File: libraries/aws_api_gateway_deployment.rb (hunk 1 above)
Comment (soumyo13) [reply libraries/aws_api_gateway_deployment.rb:16]: Removed

File: docs/resources/aws_api_gateway_deployment.md (hunk 2 above)
Comment (soumyo13) [reply docs/resources/aws_api_gateway_deployment.md:1]: Ok.

File: docs/resources/aws_api_gateway_deployment.md (hunk 3 above)
Comment (soumyo13) [reply docs/resources/aws_api_gateway_deployment.md:81]: Changed

**Review by IanMadd — APPROVED** [review]

**Review by Nirbhay1997 — APPROVED** [review]

**Pull request merged**